src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -3329,7 +3329,7 @@ public async Task<NodeConnectionTestResult> TestConnectionToRemote(string url, s
 
                 using (var cts = new CancellationTokenSource(Server.Configuration.Cluster.OperationTimeout.AsTimeSpan))
                 {
-                    connectionInfo = ReplicationUtils.GetTcpInfoAsync(url, database, "Test-Connection", Server.Certificate.Certificate,
+                    connectionInfo = ReplicationUtils.GetTcpInfoAsync(GetNodeHttpServerUrl(), url, database, "Test-Connection", Server.Certificate.Certificate,
```

Comment:
this is used by cluster (add node), I think for cluster you do not want to send 'senderUrl', please verify (because forExternalUse in that case should be false)

Reply:
verified :-)